test/textformatter.cpp

```diff
@@ -47,6 +47,35 @@ TEST_CASE("lines marked as `wrappable` are wrapped to fit width",
 	}
 }
 
+TEST_CASE("line wrapping works for non-space-separeted text",
+          "[textformatter]") {
+
+	textformatter fmt;
+
+	fmt.add_lines(
+		{
+			std::make_pair(LineType::wrappable,
+					"    つれづれなるままに、ひぐらしすずりにむかいて、")
+		});
+
+
+	SECTION("preserve indent and doesn't return broken UTF-8") {
+		const std::string expected = (
+				"    つれづれなるまま\n"
+				"    に、ひぐらしすず\n"
+				"    りにむかいて、\n");
+		REQUIRE(fmt.format_text_plain(20) == expected);
+		// +1 is not enough to store single wide-width char
+		REQUIRE(fmt.format_text_plain(20+1) == expected);
+	}
+
+	SECTION("discard a current word if no enough space is available") {
```

Comment:
Still not entirely happy about this behaviour, but it's my bugs, not yours. I filed #82 as a reminder to myself to get back to that.